concordium-contracts-common/src/schema_json.rs

```diff
@@ -956,13 +956,234 @@ impl Fields {
     }
 }
 
-/// Displays a pretty-printed JSON-template of the schema.
+/// Displays a pretty-printed JSON-template of the [`SchemaType`].
 impl Display for Type {
     fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
         write!(f, "{}", serde_json::to_string_pretty(&self.to_json_template()).unwrap())
     }
 }
 
+/// Displays the json template of the type indented.
+fn display_json_template_indented(
```

Comment:
Could you document this method a bit more?
How will it be formatted and what do the parameters mean exactly? I see that `2` is always added to `indent` in the second call of `format` which seems a bit odd if you give a large `indent` value.

Reply:
Added the following:
```
/// Displays the json template of the type indented.
/// The `type_schema_name` is indented by `indent` and the associated `type_schema` is indented by `indent + 2`.
/// The `type_schema` is one level deeper than the `type_schema_name` and therefore indented by `+ 2`.
/// '\n' is replaced by "\n        " because the `VersionedModuleSchema` template has several higher level outputs (e.g. `contractNames`, `functionNames`, ...).
```